neo/SmartContract/Native/Tokens/GasToken.cs

```diff
@@ -7,8 +16,54 @@ public sealed class GasToken : Nep5Token<Nep5AccountState>
         public override string Symbol => "gas";
         public override int Decimals => 8;
 
+        private const byte Prefix_SystemFeeAmount = 15;
+
         internal GasToken()
         {
         }
+
+        protected override StackItem Main(ApplicationEngine engine, string operation, VMArray args)
+        {
+            switch (operation)
+            {
+                case "distributeFees":
+                    return DistributeFees(engine);
+                case "getSysFeeAmount":
+                    return GetSysFeeAmount(engine, (uint)args[0].GetBigInteger());
+                default:
+                    return base.Main(engine, operation, args);
+            }
+        }
+
+        private bool DistributeFees(ApplicationEngine engine)
+        {
+            if (engine.Trigger != TriggerType.System) throw new InvalidOperationException();
+            foreach (Transaction tx in engine.Snapshot.PersistingBlock.Transactions)
+                Burn(engine, tx.Sender, tx.Gas + tx.NetworkFee);
+            ECPoint[] validators;
+            if (engine.Snapshot.PersistingBlock.Index > 1)
+                validators = engine.Snapshot.NextValidators.Get().Validators;
+            else
+                validators = Blockchain.StandbyValidators;
+            UInt160 primary = Contract.CreateSignatureRedeemScript(validators[engine.Snapshot.PersistingBlock.ConsensusData.PrimaryIndex]).ToScriptHash();
+            Mint(engine, primary, engine.Snapshot.PersistingBlock.Transactions.Sum(p => p.NetworkFee));
+            BigInteger sys_fee = GetSysFeeAmount(engine, engine.Snapshot.PersistingBlock.Index - 1) + engine.Snapshot.PersistingBlock.Transactions.Sum(p => p.Gas);
+            StorageKey key = CreateStorageKey(Prefix_SystemFeeAmount, BitConverter.GetBytes(engine.Snapshot.PersistingBlock.Index));
```

Comment:
We need to store this?

Reply:
Yes, we need this for gas distribution.

https://github.com/neo-project/neo/blob/2813c10f95571b350d6642d6ee47aaa891490b84/neo/SmartContract/Native/Tokens/NeoToken.cs#L110